src/state/apis/swappers/swappersApi.ts

```diff
@@ -50,25 +25,57 @@ export const swappersApi = createApi({
     getTradeQuote: build.query<ApiQuote[], GetTradeQuoteInput>({
       queryFn: async (getTradeQuoteInput: GetTradeQuoteInput, { dispatch, getState }) => {
         const state = getState() as ReduxState
-        const { sendAddress, receiveAddress } = getTradeQuoteInput
+        const {
+          sendAddress,
+          receiveAddress,
+          sellAsset,
+          buyAsset,
+          affiliateBps,
+          sellAmountIncludingProtocolFeesCryptoBaseUnit,
+        } = getTradeQuoteInput
         const isCrossAccountTrade = sendAddress !== receiveAddress
         const featureFlags: FeatureFlags = selectFeatureFlags(state)
+        const sellAssetUsdRate = selectUsdRateByAssetId(state, getTradeQuoteInput.sellAsset.assetId)
         const enabledSwappers = getEnabledSwappers(featureFlags, isCrossAccountTrade)
 
+        if (!sellAssetUsdRate) throw Error('missing sellAssetUsdRate')
+
         // Await market data fetching thunk, to ensure we can display some USD rate and don't bail in getDependencies above
-        await dispatch(
-          marketApi.endpoints.findByAssetId.initiate(getTradeQuoteInput.sellAsset.assetId),
+        await dispatch(marketApi.endpoints.findByAssetId.initiate(sellAsset.assetId))
+        await dispatch(marketApi.endpoints.findByAssetId.initiate(buyAsset.assetId))
+
+        const sellAmountBeforeFeesCryptoPrecision = fromBaseUnit(
+          sellAmountIncludingProtocolFeesCryptoBaseUnit,
+          sellAsset.precision,
         )
-        await dispatch(
-          marketApi.endpoints.findByAssetId.initiate(getTradeQuoteInput.buyAsset.assetId),
+        const sellAmountBeforeFeesUsd = bnOrZero(sellAmountBeforeFeesCryptoPrecision).times(
+          sellAssetUsdRate,
+        )
+        // We use the sell amount so we don't have to make 2 network requests, as the receive amount requires a quote
```

Comment:
Nice. This logic works at a request level, but the UI will still show as donating when `isDonationAmountBelowMinimum` and the quote request contains no donation.

We'll want to hide the donation button in these cases, which matches the current behaviour.

Reply:
Not sure I follow, is the UI displaying 0 donations in these cases?

Reply:
Might be resolved in following PR as I recall fixing this